server/artemis/local.ts

```diff
@@ -0,0 +1,165 @@
+/**
+ * Typescript wrapper around the Python Artemis model.
+ * By Mia.
+ * @author mia-pi-git
+ */
+
+import * as child_process from 'child_process';
+import {ProcessManager, Streams, Utils, Repl} from '../../lib';
+import {Config} from '../config-loader';
+import {toID} from '../../sim/dex-data';
+
+class ArtemisStream extends Streams.ObjectReadWriteStream<string> {
+	tasks = new Set<string>();
+	private process: child_process.ChildProcessWithoutNullStreams;
+	constructor() {
+		super();
+		this.process = child_process.spawn('python3', [
+			'-u', __dirname + '/model.py', Config.debugartemisprocesses ? "debug" : "",
+		].filter(Boolean));
+		this.listen();
+	}
+	listen() {
+		this.process.stdout.setEncoding('utf8');
+		this.process.stderr.setEncoding('utf8');
+		this.process.stdout.on('data', (data) => {
+			// so many bugs were created by \nready\n
+			data = data.trim();
+			const [taskId, dataStr] = data.split("|");
+			if (this.tasks.has(taskId)) {
+				this.tasks.delete(taskId);
+				return this.push(`${taskId}\n${dataStr}`);
+			}
+			if (taskId === 'error') { // there was a major crash and the script is no longer running
+				const info = JSON.parse(dataStr);
+				Monitor.crashlog(new Error(info.error), "An Artemis script", info);
+				try {
+					this.pushEnd(); // push end first so the stream always closes
+					this.process.disconnect();
+				} catch {}
+			}
+		});
+		this.process.stderr.on('data', data => {
+			data += "";
```

Comment:
?

Reply:
data is a Buffer by default. This just makes it into a string.

Reply:
Why choose `+= ""` instead of `.toString()`?

Reply:
Looks nicer, in my opinion.

Reply:
I think it makes the code more difficult to read for people who aren't familiar with JavaScript's type coercion.

Reply:
We can junk it anyway since setEncoding('utf-8') means the data is a string. This was just a prior relic of that.